specification.md

```diff
@@ -1642,6 +1646,50 @@ In this case the `failOnValidationErrors` default value of `true` is assumed.
 The `dataInputSchema` property validates the [workflow data input](#Workflow-Data-Input). In case of 
 a starting [Event state](#Event-state), it is not used to validate its event payloads.
 
+The `secrets` property allows you to use sensitive information such as passwords, OAuth tokens, ssh keys, etc. inside your
+Workflow expressions. 
+
+It has two possible types, `string` or `array`. 
+If `string` type, it is an URI pointing to a JSON or YAML document
+which contains an array of names of the secrets, for example:
+
+```json
+"secrets": "file://workflowsecrets.json"
+```
+
+If `array` type, it defines an array (of string types) which contains the names of the secrets, for example:
+
+```json
+"secrets": ["MY_PASSWORD", "MY_STORAGE_KEY", "MY_ACCOUNT"]
+```
+
+For more information about Workflow secrets, reference the [Workflow Secrets section](#Workflow-Secrets).
+
+The `globals` property can be used to define Workflow global values
+which are accessible in [Workflow Expressions](#Workflow-Expressions) via their defined Namespace.
+
+It has two possible types, `string` or `array`.
+If `string` type, it is an URI pointing to a JSON or YAML document
+which contains an array of global definitions, for example:
+
+```json
+"globals": "file://workflowglobals.json"
+```
+
+If `array` type, it defines an array which contains the global definitions, for example:
+
+```json
+"globals": [
+ {
+   "namespace": "AGE",
```

Comment:
Is there a functional purpose for the namespace other than organizing similar globals into the same namespace?

This seems like overkill to have "namespaces", do we really think that a single workflow definition will have multiple namespaces?   If it's just for organizination, then users that want to group values could just prefix the names of their globals (ie; `AGE.MIN_ADULT`) instead?

Reply:
for large global data it is a nice way of grouping things. also for editors that can provide auto-completion
its much easier to do like 

$TRANSLATIONS.  --> and get a list of globals that fall under that namespace than doing

$ --> and get a list of a million strings to scroll a mile through

Reply:
we could make namespace optional (blank by default) which then would be the top-level namespace? that would be an option but im not for removing it

Reply:
we're accessing all state data by their path. I think introducing namespacing in json/yaml like it's available in XML is interesting, but I think this shouldn't be a globals-specific thing. Objects are fine for grouping data, no?

Reply:
@manuelstein Yeah, agreed, and it might even be easier to integrate with jq, using --argjson flag

Reply:
ok will update for it to just be json object